src/compiler/compile.js

```diff
@@ -716,6 +717,11 @@ function compileDirectives (attrs, options) {
         pushDir(dirName, internalDirectives[dirName])
       } else {
         arg = dirName
+        // support for shorthand expression (#2843)
+        // <div v-bind:content> => <div v-bind:content="content">
+        if (!value) {
```

Comment:
Are you sure about this? We don't want this to happen when the value is `null` or `undefined`?